### PR TITLE
Implementing preflight(),loadWithErrorin NSBundle

### PR DESCRIPTION
**Pull request by saiHemak**
Also includes the fix for segmentation fault issue when the NSBundle is created with an invalid path

### CODE REVIEW

File: Foundation/NSBundle.swift [hunk 1]
```diff
@@ -73,8 +76,27 @@ public class Bundle: NSObject {
     }
     public func unload() -> Bool { NSUnimplemented() }
     
-    public func preflight() throws { NSUnimplemented() }
-    public func loadAndReturnError() throws { NSUnimplemented() }
+    public func preflight() throws {
+        var unmangaedError:Unmanaged<CFError>? = nil
```
Comment (ianpartridge) [inline Foundation/NSBundle.swift]: Typo.

File: Foundation/NSBundle.swift [hunk 2]
```diff
@@ -73,8 +76,27 @@ public class Bundle: NSObject {
...
+        try withUnsafeMutablePointer(&unmangaedError) { (unmanagedCFError: UnsafeMutablePointer<Unmanaged<CFError>?>)  in
+            CFBundlePreflightExecutable(_bundle,  unmanagedCFError)
```
Comment (ianpartridge) [inline Foundation/NSBundle.swift]: Extra space.

File: Foundation/NSBundle.swift [hunk 3]
```diff
@@ -73,8 +76,27 @@ public class Bundle: NSObject {
...
+            if let error = unmanagedCFError.pointee {
+                throw   error.takeRetainedValue()._nsObject
```
Comment (ianpartridge) [inline Foundation/NSBundle.swift]: Extra space.

File: Foundation/NSBundle.swift [hunk 4]
```diff
@@ -73,8 +76,27 @@ public class Bundle: NSObject {
...
+            }
+        }
+    }
+    
+    public func loadAndReturnError() throws {
+        var unmangaedError:Unmanaged<CFError>? = nil
```
Comment (ianpartridge) [inline Foundation/NSBundle.swift]: Typo.

File: Foundation/NSBundle.swift [hunk 5]
```diff
@@ -37,6 +37,9 @@ public class Bundle: NSObject {
         
         let url = URL(fileURLWithPath: resolvedPath)
         _bundle = CFBundleCreate(kCFAllocatorSystemDefault, unsafeBitCast(url, to: CFURL.self))
+        if (_bundle == nil) {
```
Comment (ianpartridge) [inline Foundation/NSBundle.swift]: Parentheses?

File: TestFoundation/TestNSBundle.swift [hunk 6]
```diff
@@ -29,6 +29,10 @@ class TestNSBundle : XCTestCase {
             ("test_localizations", test_localizations),
             ("test_URLsForResourcesWithExtension", test_URLsForResourcesWithExtension),
             ("test_bundleLoad", test_bundleLoad),
+            ("test_bundleLoadWithError",test_bundleLoadWithError),
+            ("test_bundleWithInvalidPath",test_bundleWithInvalidPath),
+            ("test_bundlePreflight",test_bundlePreflight),
+
```
Comment (ianpartridge) [inline TestFoundation/TestNSBundle.swift]: Extra line?

File: TestFoundation/TestNSBundle.swift [hunk 7]
```diff
@@ -29,6 +29,10 @@ class TestNSBundle : XCTestCase {
             ("test_localizations", test_localizations),
             ("test_URLsForResourcesWithExtension", test_URLsForResourcesWithExtension),
             ("test_bundleLoad", test_bundleLoad),
+            ("test_bundleLoadWithError",test_bundleLoadWithError),
```
Comment (ianpartridge) [inline TestFoundation/TestNSBundle.swift]: Space needed before function name.

**Comment by saiHemak** [comment]
@ianpartridge  Changes are done as per your comments

**Comment by parkera** [comment]
@swift-ci please test

**Pull request merged**